Mathlib/RingTheory/MvPolynomial/MonomialOrder.lean

```diff
@@ -111,6 +165,14 @@ theorem degree_X [Nontrivial R] {s : σ} :
   change m.degree (monomial (Finsupp.single s 1) (1 : R)) = _
   rw [degree_monomial, if_neg one_ne_zero]
 
+@[simp] theorem degree_one : m.degree (1 : MvPolynomial σ R) = 0 := by
+  obtain h | h' := subsingleton_or_nontrivial R
+  · rw [Subsingleton.eq_zero 1, degree_zero]
+  · change m.degree (monomial 0 1) = 0
+    classical
+    rw [degree_monomial]
+    simp
```

Comment:
(too late) Since you already put `degree_subsingleton` as `@[simp, nontriviality]`, the first part of the proof could be `exact degree_subsingleton` or `simp`, or just replace `obtain` part by `nontriviality R`.

Reply:
Feel free to open another PR!

Reply:
#22134